osquery/tables/system/darwin/smc_keys.cpp

```diff
@@ -490,5 +496,73 @@ QueryData getTemperatures(QueryContext &context) {
 
   return results;
 }
+
+QueryData getFanSpeeds(QueryContext &context) {
+  QueryData results;
+
+  SMCHelper smc;
+  if (!smc.open()) {
+    return {};
+  }
+
+  // Get number of fans.
+  QueryData key_data;
+  genSMCKey("FNum", smc, key_data);
+  if (key_data.empty()) {
+    // The SMC search for key information failed.
+    return results;
+  }
+
+  auto &smcRow = key_data.back();
+  if (smcRow["value"].empty()) {
+    return results;
+  }
+
+  // Get attributes for each fan.
+  int numFans = std::stoi(smcRow["value"]);
+  for (int fanIdx = 0; fanIdx < numFans; fanIdx++) {
+    Row r;
+    r["fan"] = std::to_string(fanIdx);
+
+    for (const auto &smcFanSpeedKey : kSMCFanSpeeds) {
+      std::stringstream key;
```

Comment:
Maybe move the `r[smcFanSpeedKey.second] = INTEGER(0);` here, then each error case can happily continue without care :dancers: